packages/malloy/src/lang/ast/query-properties/nest.ts

```diff
@@ -266,21 +274,68 @@ export class NestReference
       if (expressionIsAnalytic(type.expressionType)) {
         useInstead = 'a calculate';
         kind = 'an analytic';
-      } else if (expressionIsScalar(type.expressionType)) {
-        useInstead = 'a group_by or select';
-        kind = 'a scalar';
-      } else if (expressionIsAggregate(type.expressionType)) {
-        useInstead = 'an aggregate';
-        kind = 'an aggregate';
       } else {
-        throw new Error(`Unexpected expression type ${type} not handled here`);
+        if (this.inExperiment('scalar_lenses', true)) {
+          return;
+        }
+        if (expressionIsScalar(type.expressionType)) {
+          useInstead = 'a group_by or select';
+          kind = 'a scalar';
+        } else if (expressionIsAggregate(type.expressionType)) {
+          useInstead = 'an aggregate';
+          kind = 'an aggregate';
+        } else {
+          throw new Error(
+            `Unexpected expression type ${type} not handled here`
+          );
+        }
+        this.log(
+          `Cannot use ${kind} field in a nest operation, did you mean to use ${useInstead} operation instead?`
+        );
       }
-      this.log(
-        `Cannot use ${kind} field in a nest operation, did you mean to use ${useInstead} operation instead?`
-      );
     }
   }
 
+  makeEntry(fs: DynamicSpace): void {
+    if (fs instanceof QuerySpace) {
```

Comment:
this looks so much like the code in expand turtle i wonder if some of this logic needs to get pushed up into spacefield or queryfieldspace